pkg/admissioncontroller/webhook/auth/seed/graph/eventhandler_shoot.go

```diff
@@ -136,6 +137,18 @@ func (g *graph) handleShootCreateOrUpdate(ctx context.Context, shoot *gardencore
 		g.addEdge(exposureClassVertex, shootVertex)
 	}
 
+	// TODO(tobschli): There needs to be a better way to do this, right?
+	if caConfig := shoot.Spec.Kubernetes.ClusterAutoscaler; caConfig != nil {
+		if expanderConfig := caConfig.ExpanderConfig; expanderConfig != nil {
+			if configMapName := expanderConfig.ConfigMapName; len(configMapName) != 0 {
+				if shoot.Status.SeedName != nil {
+					configMapVertex := g.getOrCreateVertex(VertexTypeConfigMap, shoot.Namespace, configMapName)
+					g.addEdge(configMapVertex, shootVertex)
+				}
+			}
+		}
+	}
```

Comment:
Well, why not simply like this? You should ensure via validation that the `ConfigMapName` has `length > 0` when provided.

```suggestion
	if shoot.Spec.Kubernetes.ClusterAutoscaler != nil && shoot.Spec.Kubernetes.ClusterAutoscaler.ExpanderConfig != nil {
		configMapVertex := g.getOrCreateVertex(VertexTypeConfigMap, shoot.Namespace, configMapName)
		g.addEdge(configMapVertex, shootVertex)
	}
```

I don't understand the `shoot.Status.SeedName != nil` condition. Can you elaborate?